src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -2007,6 +2090,59 @@ public void onWakeUp()
         {
             homeBuilding.onWakeUp();
         }
+
+        setIsAsleep(false);
+        dataManager.set(DATA_BED_POS, new BlockPos(0,0,0));
+    }
+
+    /**
+     * Is the citizen a sleep?
+     *
+     * @return true when a sleep.
+     */
+    public boolean isAsleep()
+    {
+        return dataManager.get(DATA_IS_ASLEEP);
+    }
+
+    /**
+     * Sets if the citizen is a sleep.
+     * Caution: Use trySleep(BlockPos) for better control
+     *
+     * @param isAsleep True to make the citizen sleep.
+     */
+    public void setIsAsleep(final boolean isAsleep)
+    {
+        dataManager.set(DATA_IS_ASLEEP, isAsleep);
+    }
+
+    /**
+     * Returns the orientation of the bed in degrees.
+     */
+    @SideOnly(Side.CLIENT)
+    public float getBedOrientationInDegrees()
+    {
+        final IBlockState state = this.getBedLocation() == null ? null : this.world.getBlockState(getBedLocation());
+        if (state != null && state.getBlock().isBed(state, world, getBedLocation(), this))
+        {
+            final EnumFacing enumfacing = state.getBlock().getBedDirection(state, world, getBedLocation());
+
+            switch (enumfacing)
+            {
+                case SOUTH:
+                    return NINETY_DEGREE;
+                case WEST:
+                    return 0.0f;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Upper-case this literal "f" suffix. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS818)